chrome-launcher.ts

```diff
@@ -52,9 +52,12 @@ export interface ModuleOverrides {
 }
 
 const sigintListener = async () => {
-  for (const instance of instances) {
+  // Since sets are not iterable below es2015 we need
+  // too `.forEach` iterate over the set.
+  // https://github.com/Microsoft/TypeScript/issues/6842
+  instances.forEach(async (instance) => {
```

Comment:
won't forEach not wait for the instances to be killed though?

seems like we might need a `await Promise.all(instances.map(i => i.kill())` instead

Reply:
since we are awaiting it will :)

example :taco: 

https://www.typescriptlang.org/play/#src=const%20arr%20%3D%20%5B1%2C%202%2C%203%2C%204%5D%3B%0D%0A%0D%0Aasync%20function%20doWork(count)%20%7B%0D%0A%20%20%20%20return%20new%20Promise((res%2C%20_)%20%3D%3E%20%7B%0D%0A%20%20%20%20%20%20%20%20setTimeout(()%20%3D%3E%20res()%2C%20count)%3B%0D%0A%20%20%20%20%7D)%3B%0D%0A%7D%0D%0Aasync%20function%20wow()%20%7B%0D%0A%20%20%20%20arr.forEach(async%20(v)%20%3D%3E%20%7B%0D%0A%20%20%20%20%20%20%20%20await%20doWork(v%20*%201000)%3B%0D%0A%20%20%20%20%20%20%20%20console.log('done%20with'%2C%20v)%3B%0D%0A%20%20%20%20%7D)%3B%0D%0A%7D%0D%0A%0D%0A(async%20()%20%3D%3E%20%7B%0D%0A%20%20%20%20await%20wow()%3B%0D%0A%20%20%20%20console.log('all%20done')%3B%0D%0A%7D)()%0D%0A

Reply:
hm this example seems to prove my point, unless there's something I'm missing?

running that script prints

```
all done
done with 1
done with 2
done with 3
done with 4
```

when we'd want it to be

```
done with 1
done with 2
done with 3
done with 4
all done
```

and the generated code shows that `wow` just returns immediately after the forEach without awaiting the contents of the loop

```js
return __generator(this, function (_a) {
  // this forEach is not being awaited in any way, the return value is pretty much thrown away here
  arr.forEach(function (v) { return __awaiter(_this, void 0, void 0, function () {
      return __generator(this, function (_a) {
          switch (_a.label) {
              case 0: return [4 /*yield*/, doWork(v * 1000)];
              case 1:
                  _a.sent();
                  console.log('done with', v);
                  return [2 /*return*/];
          }
      });
  }); });
  return [2 /*return*/]; // so it just exits synchronously here
});
```

Reply:
oh or a third option, we were making different points :P I follow what you are looking for, I will address it.

Thanks